mlir/lib/IR/AsmPrinter.cpp

```diff
@@ -2556,11 +2583,30 @@ void AsmPrinter::Impl::printOptionalAttrDict(ArrayRef<NamedAttribute> attrs,
     // Print the 'attributes' keyword if necessary.
     if (withKeyword)
       os << " attributes";
+    os << " {";
+
+    SmallString<16> separator = StringRef(", ");
+    if (printerFlags.getNewlineAfterAttrLimit() &&
+        attrs.size() > *printerFlags.getNewlineAfterAttrLimit()) {
+
+      // Increase indent to match the visually match the "{ " below.
+      //currentIndent += 2;
+
+      separator.clear();
+      separator.reserve(currentIndent + 2);
+      separator.append(",\n");
+      for (size_t i = 0; i < currentIndent; ++i)
+        separator.push_back(' ');
+
+      // Already put the first attribute on its own line.
+      os << "\n";
+      os.indent(currentIndent);
+    }
 
     // Otherwise, print them all out in braces.
-    os << " {";
-    interleaveComma(filteredAttrs,
-                    [&](NamedAttribute attr) { printNamedAttribute(attr); });
+    interleave(
```

Comment:
NIT: why change line 1945 from `interleave` to `llvm::interleave` but use it unqualified here?

Reply:
This interleave here is calling the new member function that I added. 
The other interleave was calling `llvm::interleave` before, but now needs to be qualified to not call the new member function instead.